File: C/200-300/AR_242.c
#include <stdio.h>
#include <stdlib.h>


// Pre: 1
long long int erasmus(const int* v, int size) {
    int i;

    long long int ret=0;
    
    if (size!=1) {
        long long int sum=v[size-1];
        for (i=size-2;i>=0;i--) {
            ret=ret+v[i]*sum;
            sum=sum+v[i];
        }
    }

    return ret;
}
// Pos: r = Sum k: 0 <= k < v.size(): v[k]

int resuelveCaso() {
    int i;

    int n;
    scanf("%d", &n);
    if (n==0) return 0;

    int* v=(int*)malloc(n*sizeof(int));
    for (i=0;i<n;i++) scanf("%d", &v[i]);

    
    printf("%lld\n", erasmus(v, n));

    free(v);
    return 1;
}

int main() {
    while (resuelveCaso());

    return 0;
}

File: Java/200-300/AR_242.java
import java.io.*;
import java.util.*;

public class AR_242 {
    
    // Pre: true
    public static long erasmus(int[] v) {
        long ret =0;
        int n    =v.length;
        
        if (n!=1) {
            long sum=v[n-1];
            for (int i=n-2;i>=0;i--) {
                ret+=(long)v[i]*sum;
                sum+=v[i];
            }
        }

        return ret;
    }
    // Pos: r = Sum k: 0 <= k < v.size(): v[k]
    
    public static boolean resuelveCaso(BufferedReader br, BufferedWriter bw) throws IOException {
        StringTokenizer st=new StringTokenizer(br.readLine());
        int n=Integer.parseInt(st.nextToken());
        if (n==0) return false;

        int[] v=new int[n];
        st=new StringTokenizer(br.readLine());
        for (int i=0;i<n;i++) v[i]=Integer.parseInt(st.nextToken());

        bw.write(Long.toString(erasmus(v)));
        bw.newLine();

        return true;
    }

    // Usando buffered en vez de scanner se consigue reducir el tiempo de ejecucion
    // considerablemente.
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
        

        while (resuelveCaso(br, bw));

        bw.flush();
        bw.close();
        br.close();
    }
}